zbm-builder.sh

```diff
@@ -1,70 +1,145 @@
+    exit 1
+  fi
+}
+
+sanitise_file() {
+  if [ -x "${OPTARG}" ] && ! [ -d "${OPTARG}" ]; then
+    OPTARG=$(readlink -f "${OPTARG}")
+  else
+    echo "Error: ${OPTARG} is not a valid, executable file."
+    exit 1
+  fi
+}
+
 usage() {
-  cat <<EOF
+  cat << EOF
+
+zbm-builder.sh
+
+This script generates bootable ZFSBootMenu components inside a pre-made
+build container, freeing the user from having to install ZFSBootMenu and
+dependencies on the host system or invoke the build scripts directly.
+
+For more information, please see BUILD.md and README.md, or find the latest
+documentation at https://github.com/zbm-dev/zfsbootmenu/
+
 Usage: $0 [options]
+
   OPTIONS:
-  -h Display help text
+ 
+  -1 <file>
+     Fetch inidcated file to build directory and mark for integration as
+     a 'zfsbootmenu_early_setup' hook. This command can be invoked multiple
+     times.
+
+  -2 <file>
+     Fetch inidcated file to build directory and mark for integration as
+     a 'zfsbootmenu_setup' hook. This comand can be invoked multiple times.
+  
+  -3 <file>
+     Fetch inidcated file to build directory and mark for integration as
+     a 'zfsbootmenu_teardown' hook. This command can be invoked multiple
+     times.
 
-  -L Use local './' source tree instead of remote.
-     (Default: upstream master.)
+  -b <path>
+     Use an alternate build directory for the container.
+     (By default, the directory containing 'zbm-builder.sh' is used.)
 
-  -H Do not copy /etc/hostid into image
-     (Has no effect if ./hostid exists)
+  -d Force use of Docker instead of Podman as container management tool.
 
-  -C Do not copy /etc/zfs/zpool.cache into image
-     (Has no effect if ./zpool.cache exists)
+  -h Display this help text.
 
-  -d Force use of docker instead of podman
+  -i <image>
+     Use a different container image or version as build environment.
+     (By default, the official ghcr.io/zbm-dev/zbm-builder:latest is used.)
+
+  -l <path>
+     Use local ZFSBootMenu source tree at <path> instead of remote repository.
+     (By default, a source archive is pulled by the container from GitHub.)
 
   -t <tag>
-     Build the given ZFSBootMenu tag
-     (Default: upstream master)
+     Build a specific ZFSBootMenu release tag (e.g. v1.12.0, v1.10.1, v1.9.0).
+     (By default, the current upstream master is used.)
 
-  -B <tag>
-     Use the given zbm-builder tag to build images
-     (Default: ghcr.io/zbm-dev/zbm-builder:latest)
-EOF
-}
+  -C Do not integrate /etc/zfs/zpool.cache into build image image from host.
+     (If ./zpool.cache exists, this switch will be ignored.)
 
-if command -v podman >/dev/null 2>&1; then
-  PODMAN=podman
-elif command -v docker >/dev/null 2>&1; then
-  PODMAN=docker
-fi
+  -H Do not integrate /etc/hostid into build image from host.
+     (If ./hostid exists, this switch will be ignored.)
 
+EOF
+}
+  
 SKIP_HOSTID=
 SKIP_CACHE=
 
 BUILD_TAG="ghcr.io/zbm-dev/zbm-builder:latest"
+BUILD_DIRECTORY=
```

Comment:
The build directory should have a sensible default and should be sanitized after all argument processing. Also, the local repo mount should be handled similarly, so
```suggestion
BUILD_DIRECTORY="."
LOCAL_REPO=""
```
Here, `$LOCAL_REPO` may remain empty to indicate that no local mount is desired, but a local build directory will always be used.

Reply:
:+1: